### DISCUSSION TITLE
why it doesnt work on tasker and Knock on ports

### DISCUSSION THREAD

**Discussion by iamsad3508** [shadowsocks/shadowsocks-android#3032] [Q&A]
Device: Pixel 5, Android version 13. SS version should be at least 5.3.3. 

The issue is that with global proxy or bypassing local network or mainland addresses 
The APP: knock on ports, port knocking cannot be completed. 
THE other app:Tasker,  cannot ping a blocked IP either.

 Suspect that SS is not proxying these two apps.

**Comment by dev4u** [comment]
这个通常跟你的配置有关。

K.O.P.成功与否，是与你使用的敲门协议、到达顺序有关系！
如果之前你没接触过KOP，你需要不断调整协议、敲门端口的数量，以适应你的环境。

**Reply by iamsad3508** [reply]
如果是选择GFW列表、仅代理中国大陆地址或不运行ss，是可以打开端口的，移动数据、WIFI都一样。

端口开门用的是knockd，在运行ss的服务器上执行knockd ip 11111:tcp也是可以打开端口的

所以推测是ss“代理”了这个K.O.P，但同时又没有“代理”得到K.O.P的流量

还非常无语地开了飞行模式，K.O.P都可以提示knocking complete

**Reply by dev4u** [reply]
有个参数可以控制敲门间隔，加上试试。

**Reply by iamsad3508** [reply]
刚换了只kncock一个端口也不行

**Reply by dev4u** [reply]
如果你认为ss影响了你的场景，那就把ss关了看看是不是正常？如果关了还是不正常，那应该循其他方式解决。

如果关了恢复正常，那打开ss的日志开关，调整日志级别。重新操作后，把受影响操作的那段日志贴上来看看。

**Reply by iamsad3508** [reply]
[log.txt](https://github.com/shadowsocks/shadowsocks-android/files/11020729/log.txt)
不太会用日志，看有没有截全吧，直接调ss好像又没日志输出，倒是调knock on ports倒是有：
没开SS就是下面
03-21 00:36:34.956 25183 25206 I Knocker : Knocking to 'maiami'
03-21 00:36:34.957 25183 25206 D TrafficStats: tagSocket(78) with statsTag=0xffffffff, statsUid=-1
03-21 00:36:34.957 25183 25206 I Knocker : Knock #1
03-21 00:36:34.957 25183 25206 I TCP     : hitting 《IP》:10603
03-21 00:36:35.461 25183 25206 I Knocker : Knock #2
03-21 00:36:35.963 25183 25206 I Knocker : Knock #3
03-21 00:36:36.465 25183 25206 I Knocker : Knocking complete


开了SS就是下面
03-21 00:36:50.125 25183 25206 I Knocker : Knocking to 'maiami'
03-21 00:36:50.126 25183 25206 D TrafficStats: tagSocket(92) with statsTag=0xffffffff, statsUid=-1
03-21 00:36:50.126 25183 25206 I Knocker : Knock #1
03-21 00:36:50.127 25183 25206 I TCP     : hitting 《IP》:10603
03-21 00:36:50.629 25183 25206 I Knocker : Knock #2
03-21 00:36:51.130 25183 25206 I Knocker : Knock #3
03-21 00:36:51.632 25183 25206 I Knocker : Knocking complete

**Reply by iamsad3508** [reply]
好像没开ss的时候，也看见过没这句的，我再确认好了：TrafficStats: tagSocket(92) with statsTag=0xffffffff, statsUid=-1

日志是在只代理kop的情况下的
